python/private/pypi/dependency_resolver/dependency_resolver.py

```diff
@@ -15,14 +15,17 @@
 "Set defaults for the pip-compile command to run it under Bazel"
 
 import atexit
+import functools
 import os
 import shutil
 import sys
 from pathlib import Path
-from typing import Optional, Tuple
+from typing import List, Optional, Tuple
 
 import click
 import piptools.writer as piptools_writer
+from pip._internal.exceptions import DistributionNotFound
+from pip._vendor.resolvelib.resolvers import ResolutionImpossible
```

Comment:
I'm always wary of using internals as they're subject to change without warning. Are there any public exceptions that can be used instead?

Reply:
Unfortunately, it seems like not. I copied these imports from pip-tools:

https://github.com/jazzband/pip-tools/blob/e4ed0c1e028d1ca73673a51722ba153f0c02b0c6/piptools/resolver.py#L11

https://github.com/jazzband/pip-tools/blob/e4ed0c1e028d1ca73673a51722ba153f0c02b0c6/piptools/resolver.py#L24